libraries/Microsoft.Bot.Builder/StreamingRequestHandler.cs

```diff
@@ -0,0 +1,207 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder
+{
+    /// <summary>
+    /// Used to process incoming requests sent over an <see cref="IStreamingTransport"/> and adhering to the Bot Framework Protocol v3 with Streaming Extensions.
+    /// </summary>
+    public class StreamingRequestHandler : RequestHandler
+    {
+#if DEBUG
+        public
+#else
+        private
+#endif
+        string userAgent;
+
+        private IBot bot;
+
+        private IServiceProvider services;
+
+        private IList<IMiddleware> middlewareSet;
+
+        private Func<ITurnContext, Exception, Task> onTurnError;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamingRequestHandler"/> class.
+        /// The StreamingRequestHandler serves as a translation layer between the transport layer and bot adapter.
+        /// It receives ReceiveRequests from the transport and provides them to the bot adapter in a form
+        /// it is able to build activities out of, which are then handed to the bot itself to processed.
+        /// Throws <see cref="ArgumentNullException"/> if arguments are null.
+        /// </summary>
+        /// <param name="onTurnError">Optional function to perform on turn errors.</param>
+        /// <param name="bot">The <see cref="IBot"/> to be used for all requests to this handler.</param>
+        /// <param name="middlewareSet">An optional set of middleware to register with the bot.</param>
+        public StreamingRequestHandler(Func<ITurnContext, Exception, Task> onTurnError, IBot bot, IList<IMiddleware> middlewareSet = null)
+        {
+            this.bot = bot ?? throw new ArgumentNullException(nameof(bot));
+            this.middlewareSet = middlewareSet;
+            userAgent = GetUserAgent();
+            this.onTurnError = onTurnError;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamingRequestHandler"/> class.
+        /// An overload for use with dependency injection via ServiceProvider, as shown
+        /// in DotNet Core Bot Builder samples.
+        /// Throws <see cref="ArgumentNullException"/> if arguments are null.
+        /// </summary>
+        /// <param name="onTurnError">Optional function to perform on turn errors.</param>
+        /// <param name="serviceProvider">The service collection containing the registered IBot type.</param>
+        /// <param name="middlewareSet">An optional set of middleware to register with the bot.</param>
+        public StreamingRequestHandler(Func<ITurnContext, Exception, Task> onTurnError, IServiceProvider serviceProvider, IList<IMiddleware> middlewareSet = null)
+        {
+            services = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+            this.middlewareSet = middlewareSet;
+            userAgent = GetUserAgent();
+            this.onTurnError = onTurnError;
+        }
```

Comment:
Null check this. If null is allowed (ick!) create a default instance that does nothing, so all the code in your made methods below don't need to null check.

Reply:
I'm not sure if this is anchored to the right line, but if it is, we're specifically not null checking onTurnError because BotAdapter's RunPipelineAsync uses a null check on it to decide if it should handle exceptions or throw them.